cloudformation/aws-parallelcluster.cfn.json

```diff
@@ -2790,6 +2790,13 @@
           "MasterENI": {
             "Ref": "MasterENI"
           },
+          "MasterPublicIp": {
```

Comment:
Although this is similar to existing code and probably works, we might want to differentiate these conditions/parameter to make the code more readable.

For example, this one can be called `UseMasterPublicIpInSubstack`